packages/@uppy/status-bar/src/Components.tsx

```diff
@@ -192,10 +245,14 @@ function LoadingSpinner () {
   )
 }
 
-function ProgressBarProcessing (props) {
+interface ProgressBarProcessingProps {
+  progress: FileProcessingInfo
+}
+
+function ProgressBarProcessing(props: ProgressBarProcessingProps): JSX.Element {
   const { progress } = props
   const { value, mode, message } = progress
-  const roundedValue = Math.round(value * 100)
+  const roundedValue = Math.round(value! * 100)
```

Comment:
This PR is about adding types, we should refrain from making any change that would affect the JS output

Reply:
But I thought that rule was to not introduce breaking changes. In this case, does it matter that the JS output changes? I think we’re missing on the critical benefit of a TS refactor, which is improving our code to be more stable along the way, like in this case.

Reply:
Shouldn't we fix be fixing bugs that we discover while refactoring to TS, even though it changes the JS output? At the very least we should add a TODO to fix such discovered bugs in the future

Reply:
IMO non-null assertions are an implicit TODO. We should go through them and remove them on the 4.x branch

Reply:
I think non-null assertions are fine to use in some cases, when the alternative is to check for nullish and throw an error (like `maybeNull!.something`). But in this case, the bug will render NaN, and in this case the correct solution is probably not to throw an error, but instead render something like `0`.

Reply:
https://github.com/transloadit/uppy/blob/4854bba2bdcc4a0e6b878f670d9890d607e8b403/packages/@uppy/status-bar/src/Components.tsx#L260

As you can see, if the value is indeterminate, that `roundedValue` will not be read.

Reply:
then it's better to put the round *after* the determinate check, then we don't need the non-null assertion, because typescript knows that it's not null:
```js
      {mode === 'determinate' ? `${Math.round(value * 100)}% ${dot} ` : ''}
```

Reply:
I agree, but I'd prefer to not introduce runtime changes when we can avoid it. My wish is we'd be able to get rid of all non-null assertions on the 4.x branch, and turn the linter warning to an error.